src/apps/bridge/rbrPressureProcessor.cpp

```diff
@@ -0,0 +1,201 @@
+#include "task_priorities.h"
+#include "timers.h"
+#include <inttypes.h>
+#include <math.h>
+#include <stdlib.h>
+
+#ifndef RAW_PRESSURE_ENABLE
+#error "RAW_PRESSURE_ENABLE must be defined"
+#endif // RAW_PRESSURE_ENABLE
+
+typedef struct PressureProcessorContext {
+  QueueHandle_t q;
+  EventGroupHandle_t eg;
+  TimerHandle_t sendTimer;
+  uint32_t rawSampleS;
+  uint32_t diffBitDepth;
+  uint32_t maxRawReports;
+  double rawDepthThresholdUbar;
+  bool started;
+} PressureProcessorContext_t;
+
+typedef struct reportMetaData {
+  uint32_t magic;
+  uint32_t nRawReportsSent;
+} reportMetaData_t;
+
+static constexpr EventBits_t kQRcv = 1 << 0;
+static constexpr EventBits_t kTimEx = 1 << 1;
+static constexpr EventBits_t kAll = kQRcv | kTimEx;
+static constexpr uint32_t kRbrSamplePeriodMs = 500;
+static constexpr uint32_t kRbrSamplePeriodErrMs = 5;
+static constexpr uint32_t kNoInitMagic = 0xcafedaad;
+static constexpr char kRbrPressureProcessorTag[] = "[RbrPressureProcessor]";
+static constexpr size_t cbor_buffer_size = 1024;
+static constexpr char kRbrPressureHdrTopic[] = "/sofar/bm_rbr_data";
+static constexpr uint32_t kSampleChunkSize = 30;
+
+static void runTask(void *param);
+static void diffSigSendTimerCallback(TimerHandle_t xTimer);
+
+static PressureProcessorContext_t _ctx;
+#ifndef CI_TEST
+static reportMetaData_t _reportMetaData __attribute__((section(".noinit")));
+#else  // CI_TEST
+static reportMetaData_t _reportMetaData;
+#endif // CI_TEST
+
+void rbrPressureProcessorInit(uint32_t rawSampleS, uint32_t diffBitDepth,
+                              uint32_t maxRawReports, double rawDepthThresholdUbar) {
+  _ctx.q = xQueueCreate(10, sizeof(BmRbrDataMsg::Data));
+  configASSERT(_ctx.q);
+  _ctx.eg = xEventGroupCreate();
+  configASSERT(_ctx.eg);
+  _ctx.rawSampleS = rawSampleS;
+  _ctx.sendTimer = xTimerCreate("rbrRawSendTimer", pdMS_TO_TICKS(_ctx.rawSampleS * 1000),
+                                pdFALSE, NULL, diffSigSendTimerCallback);
+  configASSERT(_ctx.sendTimer);
+  _ctx.diffBitDepth = diffBitDepth;
+  _ctx.maxRawReports = maxRawReports;
+  _ctx.rawDepthThresholdUbar = rawDepthThresholdUbar;
+  _ctx.started = false;
+  if (_reportMetaData.magic != kNoInitMagic) {
+    _reportMetaData.magic = kNoInitMagic;
+    _reportMetaData.nRawReportsSent = 0;
+  }
+  configASSERT(xTaskCreate(runTask, "rbrPressureProcessor", configMINIMAL_STACK_SIZE * 4, NULL,
+                           RBR_PROCESSOR_TASK_PRIORITY, NULL) == pdPASS);
+}
+
+void rbrPressureProcessorStart(void) {
+  configASSERT(xTimerStart(_ctx.sendTimer, 10) == pdPASS);
+  _ctx.started = true;
+}
+
+bool rbrPressureProcessorAddSample(BmRbrDataMsg::Data &rbr_data, uint32_t timeout_ms) {
+  if (_ctx.started && xQueueSend(_ctx.q, &rbr_data, pdMS_TO_TICKS(timeout_ms)) == pdTRUE) {
+    xEventGroupSetBits(_ctx.eg, kQRcv);
+    return true;
+  }
+  return false;
+}
+
+bool rbrPressureProcessorIsStarted(void) { return _ctx.started; }
+
+static void runTask(void *param) {
```

Comment:
A loop waiting for two events:
1. the RBR driver submitting a sample to be put intto the diff signal
2. the Sample timer expiring, at which point we should send to Spotter.